lib/ansible/modules/storage/netapp/na_ontap_ldap_client.py

```diff
@@ -0,0 +1,365 @@
+            ldap_servers=dict(required=False, type='list'),
+            min_bind_level=dict(required=False, default=None, choices=['anonymous', 'simple', 'sasl']),
+            port=dict(required=False, default=None, type='int'),
+            query_timeout=dict(required=False, default=None, type='int'),
+            referral_enabled=dict(required=False, default=None, choices=['true', 'false']),
+            schema=dict(required=False, default=None, type='str', choices=['AD-IDMU', 'AD-SFU', 'MS-AD-BIS', 'RFC-2307']),
+            session_security=dict(required=False, default=None, choices=['true', 'false']),
+            state=dict(required=False, choices=['present', 'absent'], default='present'),
+            use_start_tls=dict(required=False, default=None, choices=['true', 'false']),
+            vserver=dict(required=True, type='str')
+        ))
+
+        self.module = AnsibleModule(
+            argument_spec=self.argument_spec,
+            supports_check_mode=True
+        )
+        self.na_helper = NetAppModule()
+        self.parameters = self.na_helper.set_parameters(self.module.params)
+
+        if HAS_NETAPP_LIB is False:
+            self.module.fail_json(
+                msg="the python NetApp-Lib module is required")
+        else:
+            self.server = netapp_utils.setup_na_ontap_zapi(module=self.module, vserver=self.parameters['vserver'])
+
+        self.simple_attributes = [
+            'base_dn',
+            'base_scope',
+            'bind_dn',
+            'bind_password',
+            'min_bind_level',
+            'port',
+            'query_timeout',
+            'referral_enabled',
+            'session_security',
+            'use_start_tls'
+        ]
+
+        # Handle required parameters for state=present
+        if self.module.params.get('state') == 'present':
+            required_when_present = ('ldap_servers', 'schema')
+            msg = ""
+            missing_param = False
+
+            for param in required_when_present:
+                if not self.module.params.get(param):
+                    msg += "Option '" + str(param) + "' is required  when 'state' = 'present'.\n"
+                    missing_param = True
+
+            if missing_param:
+                self.module.fail_json(msg=msg)
+
+    def get_ldap_client(self, client_config_name=None):
+        '''
+        Checks if LDAP client config exists.
+
+        :return:
+            ldap client config object if found
+            None if not found
+        :rtype: object/None
+        '''
+        # Make query
+        client_config_info = netapp_utils.zapi.NaElement('ldap-client-get-iter')
+
+        if client_config_name is None:
+            client_config_name = self.parameters['client_config']
+
+        query_details = netapp_utils.zapi.NaElement.create_node_with_children('ldap-client', **{'ldap-client-config': client_config_name})
+
+        query = netapp_utils.zapi.NaElement('query')
```

Comment:
It could be safer to add vserver to the query

Reply:
I've just pushed updated code. Please check it.